File: hash_table/0884_uncommon_words_from_two_sentences/0884_uncommon_words_from_two_sentences.go
package main

import (
	"fmt"
	"strings"
)

func uncommonFromSentences(A string, B string) []string {
	words := map[string]bool{}
	ret := []string{}

	var help = func(s string) {
		for _, word := range strings.Split(s, " ") {
			_, ok := words[word]
			words[word] = !ok
		}
	}

	help(A)
	help(B)

	for key, val := range words {
		if val {
			ret = append(ret, key)
		}
	}

	return ret
}

func main() {
	A := "this apple is sweet"
	B := "this apple is sour"

	fmt.Println(uncommonFromSentences(A, B))
}

File: hash_table/0884_uncommon_words_from_two_sentences/0884_uncommon_words_from_two_sentences.py
class Solution(object):
    def uncommonFromSentences(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: List[str]
        """
        res = []
        d = {}
        for s in A.split(" ") + B.split(" "):
            if s in d:
                d[s] += 1
            else:
                d[s] = 1
        
        for k,v in d.items():
            if v == 1:
                res.append(k)

        return res


A = "this apple is sweet"
B = "this apple is sour"

print(Solution().uncommonFromSentences(A, B))